src/components/PeoplePage.tsx

```diff
@@ -0,0 +1,62 @@
+import { useEffect, useState } from 'react';
+import { getPeople } from '../api';
+import { Loader } from './Loader';
+import { Person } from '../types';
+import PeopleTable from './PeopleTable';
+import ErrorMessage from './ErrorMessage';
+import NotPeople from './NotPeople';
+
+const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[] | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+
+  useEffect(() => {
+    let isMounted = true;
+
+    setIsLoading(true);
+
+    getPeople()
+      .then(data => {
+        if (isMounted) {
+          setPeople(data);
+        }
+      })
+      .catch(() => {
+        if (isMounted) {
+          setIsError(true);
+        }
+      })
+      .finally(() => {
+        if (isMounted) {
+          setIsLoading(false);
+        }
+      });
+
+    return () => {
+      isMounted = false;
+    };
+  }, []);
+
+  return (
+    <>
+      <h1 className="title">People Page</h1>
+
+      <div className="block">
+        <div className="box table-container">
+          {isLoading ? (
+            <Loader />
+          ) : isError ? (
+            <ErrorMessage />
+          ) : !people?.length ? (
```

Comment:
Instead of checking '!people?.length', you could simply check 'people.length' if you initialize 'people' as an empty array.